### ISSUE TITLE
Loss increasing while WER decreasing when finetuning wav2vec 2.0

### ISSUE THREAD

**Issue by natspan** [pytorch/fairseq#2854]
I am finetuning a wav2vec 2.0 model and I noticed that while the validation loss is increasing after a while the WER keeps decreasing. Does that mean that my model is overfitting?

Validation Loss
![image](https://user-images.githubusercontent.com/47210947/98245143-26f49880-1f79-11eb-8a51-ad507bdaaeac.png)

Validation Viterbi WER 
![image](https://user-images.githubusercontent.com/47210947/98245184-396ed200-1f79-11eb-9e8e-a6f548dc362d.png)

**Comment by alexeib** [comment]
from what i've seen, using checkpoint with lowest lm-decoded wer (even if with small beam, etc) tends to give better lm-decoded results later when using a more powerful lm and bigger beam. one hypothesis is that as the training progresses, the model tries hard to fix errors that can be easily fixed by LM (while introducing errors that are harder for the LM to fix).

choosing checkpoints based on raw WER tends to work reasonably well, just slightly worse than the lm-decoded wer. you can also play around with masking params to see if you can train for a bit longer before seeing the phenomenon you are observing

**Comment by natspan** [comment]
@alexeib Thanks for the response, I will try masking params as you said. 
The reason I am not using the lm-decoded wer is that it always stays around 100 for my dataset. I was wondering if my phenomenon can be considered as overfitting? Should I choose a checkpoint that doesn't have the lowest wer due to the increase in loss?

**Comment by alexeib** [comment]
thats very strange that wer with lm stays at 100 while raw wer is decreasing. this implies something is wrong with your lm or setup.

as for whether to use loss or raw wer as early stopping criterion, i would probably go with raw wer

**Comment by natspan** [comment]
when i use the trained model for inference with the same lm and bigger beam it gives me correct results. I stopped using the dedoded wer as i thought that for the language that i am using it (not english) the beam=50 doesn't give good results.
